### ISSUE TITLE
Announcements are always shown twice in notifications

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/announcementcenter#781]
### Steps to reproduce
1. Admin makes an announcement
2. Open notification box

### Expected behaviour
Only 1 notification per announcement.

### Actual behaviour
2 notifications per announcement:
![image](https://github.com/nextcloud/announcementcenter/assets/925062/dc6523ee-c11e-409b-9d50-5748e9c90b91)

### Server configuration
Current cloud.nc

### Client configuration
**Browser:** Firefox 123.0

**Operating system:** Ubuntu 22.04.4

**Comment by jancborchardt** [comment]
Seems this is the same as https://github.com/nextcloud/announcementcenter/issues/637 ?

For "An announcement is posted by an administrator" push was still enabled. I disabled it now, but I would assume that if it's enabled I get 1 notification, and if I disable it I see nothing at all.

**Comment by nickvergessen** [comment]
> but I would assume that if it's enabled I get 1 notification, and if I disable it I see nothing at all.

No, that is for activities. Notifications and activities are admin options while posting. You basically enabled to get the activity mirrored into your notifications.

I will try to prevent this by forcing the push setting to off

**Comment by SimJoSt** [comment]
We believe we ran into the same issue.
Our process was like this:
1. Started creating an announcement
2. Disable creating a notification for the announcement
3. Create the announcement

Some users still received a "notification" email about the announcement, which was not our intention. They also received push notifications about the announcement.

Going through all the settings in our Nextcloud instance, I believe 2 settings were responsible for this:
- the default Activity setting  "An announcement is posted by an administrator" was enabled for "Push"
- the default Notification setting "Send email reminders about unhandled notifications after" was set to "1 week"

This led to nearly all users receiving a push notification for the announcement activity, even though the admin creating the announcement didn't intend on anyone receiving a notification, and the notification email reminder caused a second wave of email notifications 1 week later for a lot of users.

We solved our own issues by adjusting these default settings as well as the settings of users directly.

I suggest the following changes:
- only create one notification for users per announcement
  - even if the admin creates a notification AND an activity and a notification is created from the activity as a second step
  - having 2 paths that could lead to a notification is confusing
  - this would currently need to check
    - if an activity is created for the announcement
    - for which users a notification will be created by the activity
    - only create a separate notification for the users that wouldn't receive a notification from the activity
- Change the wording of the "Create notifications" option while creating an announcement
  - make it clear to the admin, that this option will overwrite the default (activity) notification scheme and will notify all users, regardless of preferences

I'm not sure what other changes, if any, should be made to the current system. Maybe adjusting the wording could already suffice.
Admins should be aware that creating announcements without enabling the "Create notifications" option, could still notify users, depending on the Activity settings.
It should be clear which setting is overwriting which other setting.

Possible changes:
- remove/disable the Activity setting "An announcement is posted by an administrator" entirely
  - the email option is globally disabled anyway
  - it would allow the admin to fully control the notification behavior of the announcement
- remove the "Create notifications" option during the creation of an announcement entirely
  - Creating notifications would rely entirely on an activity being created and the connected notification options

**Comment by SimJoSt** [comment]
Reading through this https://github.com/nextcloud/announcementcenter/issues/637 I realize I don't know if the "Create notifications" option is only creating a push notification or is sending emails as well.
If the latter is the case, my previous comment and suggestions would slightly differ.

**Comment by SimJoSt** [comment]
@nickvergessen can you explain how the fix works?
Was there a bug that is fixing what OP describes?
Or is it mitigating duplicate notifications because of announcements also potentially causing notifications?

**Comment by nickvergessen** [comment]
> Was there a bug that is fixing what OP describes?

Indirectly.

Announcements trigger notifications and activity entries.
A change in activity 2 years later allowed to get notifications for activity entries.
Now it's hard coded that you can not get notifications for activities of this app.

**Comment by SimJoSt** [comment]
Thank you for the explanation. That was the fix we also implemented by just disabling the notification for announcement activities through the database.
